x-pack/plugins/inference/server/routes/chat_complete.ts

```diff
@@ -100,13 +100,45 @@ export function registerChatCompleteRoute({
 
       const { connectorId, messages, system, toolChoice, tools, functionCalling } = request.body;
 
+      const chatCompleteResponse = await client.chatComplete({
+        connectorId,
+        messages,
+        system,
+        toolChoice,
+        tools,
+        functionCalling,
+        stream: false,
+      });
+
+      return response.ok({
+        body: chatCompleteResponse,
+      });
+    }
+  );
+  router.post(
```

Comment:
I can see why you created a dedicated endpoint for the streamed response, but I worry about accidental divergence between the two implementations. I'm ok with what you have for now, but we may want to re-evaluate if the complexity of these endpoints starts to grow.

Reply:
I do want two endpoints, because I don't think having a single path/endpoint returning different kind/shape of responses format depending on the input is a very pattern, but I can try factorizing the endpoints implementation. In practice it's more code that keeping them separate, but it solves the problem of risk of divergence.

Reply:
Would https://github.com/elastic/kibana/pull/198646/commits/253cf6e12d68b8a1836ee4063f69466cd53759b3 ease some of your concerns?

Reply:
> Would [253cf6e](https://github.com/elastic/kibana/commit/253cf6e12d68b8a1836ee4063f69466cd53759b3) ease some of your concerns?

Yes it would - thanks!